File: RxUI/src/main/java/com/tamsiree/rxui/view/progressing/sprite/RectSprite.java
package com.tamsiree.rxui.view.progressing.sprite;

import android.animation.ValueAnimator;
import android.graphics.Canvas;
import android.graphics.Paint;

/**
 * @author tamsiree
 */
public class RectSprite extends ShapeSprite {
    @Override
    public ValueAnimator onCreateAnimation() {
        return null;
    }

    @Override
    public void drawShape(Canvas canvas, Paint paint) {
        if (getDrawBounds() != null) {
            canvas.drawRect(getDrawBounds(), paint);
        }
    }
}

File: RxUI/src/main/java/com/tamsiree/rxui/view/loadingview/sprite/RectSprite.kt
package com.tamsiree.rxui.view.loadingview.sprite

import android.animation.ValueAnimator
import android.graphics.Canvas
import android.graphics.Paint

/**
 * @author tamsiree
 */
open class RectSprite : ShapeSprite() {
    override fun onCreateAnimation(): ValueAnimator? {
        return null
    }

    override fun drawShape(canvas: Canvas?, paint: Paint?) {
        if (drawBounds != null) {
            canvas?.drawRect(drawBounds, paint!!)
        }
    }
}